rcl_yaml_param_parser/src/parser.c

```diff
@@ -993,118 +989,120 @@ static rcl_ret_t parse_value(
             if (NULL != ret_val) {
               allocator.deallocate(ret_val, allocator.state);
             }
-            return RCL_RET_BAD_ALLOC;
+            return RCUTILS_RET_BAD_ALLOC;
           }
         } else {
           if (*seq_data_type != val_type) {
-            RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+            RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
               "Sequence should be of same type. Value type 'string' do not belong at line_num %d",
               line_num);
             allocator.deallocate(ret_val, allocator.state);
-            return RCL_RET_ERROR;
+            return RCUTILS_RET_ERROR;
           }
         }
-        res = add_val_to_string_arr(param_value->string_array_value, ret_val, allocator);
-        if (RCL_RET_OK != res) {
+        ret = add_val_to_string_arr(param_value->string_array_value, ret_val, allocator);
+        if (RCUTILS_RET_OK != ret) {
           if (NULL != ret_val) {
             allocator.deallocate(ret_val, allocator.state);
           }
-          return res;
+          return ret;
         }
       }
       break;
     default:
-      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
         "Unknown data type of value %s at line %d", value, line_num);
-      res = RCL_RET_ERROR;
+      ret = RCUTILS_RET_ERROR;
       break;
   }
-  return res;
+  return ret;
 }
 
 ///
 /// Parse the key part of the <key:value> pair
 ///
-static rcl_ret_t parse_key(
+static rcutils_ret_t parse_key(
   const yaml_event_t event,
   uint32_t * map_level,
   bool * is_new_map,
   namespace_tracker_t * ns_tracker,
   rcl_params_t * params_st)
 {
-  int32_t res = RCL_RET_OK;
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(map_level, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(params_st, RCUTILS_RET_INVALID_ARGUMENT);
+  rcutils_allocator_t allocator = params_st->allocator;
+  RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
+    &allocator, "invalid allocator", return RCUTILS_RET_INVALID_ARGUMENT);
+
   const size_t val_size = event.data.scalar.length;
   const char * value = (char *)event.data.scalar.value;
   const uint32_t line_num = ((uint32_t)(event.start_mark.line) + 1U);
-  size_t num_nodes;
-  size_t node_idx = 0U;
-  rcl_allocator_t allocator;
 
-  if ((NULL == map_level) || (NULL == params_st)) {
-    return RCL_RET_INVALID_ARGUMENT;
+  if (NULL == value) {
+    RCUTILS_SET_ERROR_MSG("event argument has no value");
+    return RCUTILS_RET_INVALID_ARGUMENT;
   }
```

Comment:
Use `RCUTILS_CHECK_FOR_NULL_WITH_MSG`

Reply:
Done in cf8bff1e0420edf110631ff124963d6a199f638f.